packages/vite/src/node/plugins/css.ts

```diff
@@ -361,11 +362,9 @@ export function cssPostPlugin(config: ResolvedConfig): Plugin {
       let code: string
       if (usedRE.test(id)) {
         if (inlined) {
-          code = `export default ${JSON.stringify(
-            await minifyCSS(css, config)
-          )}`
+          code = `export default \`${await minifyCSS(css, config)}\``
         } else {
-          code = modulesCode || `export default ${JSON.stringify(css)}`
+          code = modulesCode || `export default \`${css}\``
```

Comment:
When there is a css like these, it errors with `[rollup-plugin-dynamic-import-variables] Unexpected token (22:4)`.
```css
div::before {
  content: '`'
}
```
```css
/* `aaa */
```
```css
/* `aaa` */
```
```css
/* ${ */
```
So I think `` ` `` and `\` and `${` needs to be escaped. (Not sure whether it is enough)

Reply:
I would suggest a sanitizing utility to handle escaping all strings in generated code. Related: https://github.com/vitejs/vite/pull/5896.

Reply:
Agreed, it was in my TODO list for the PR to refactor the construction of template strings away 👍🏼

Reply:
Also check out https://github.com/unjs/knitwork.

Reply:
knitwork looks great, I think we should use it. It will clean up a lot of code

Reply:
Added
```
export function stringifyAsTemplateLiteral(s: string) {
  return `\`${s.replace(/`|\\|\$\{/g, '\\$&')}\``
}
```